File: java/base/src/main/java/vsdk/toolkit/io/context/ReaderContext.java
package vsdk.toolkit.io.context;

import java.io.InputStream;

public class ReaderContext {
    public static final int MGF_MAXIMUM_INPUT_LINE_LENGTH = 4096;
    public static final int MGF_MAXIMUM_ARGUMENT_COUNT = (MGF_MAXIMUM_INPUT_LINE_LENGTH / 4);

    public String fileName;
    public InputStream inputStream; // stream pointer
    public int fileContextId;
    public String inputLine;
    public int lineNumber;
    public char isPipe; // Flag indicating whether input comes from a pipe or a real file
    public ReaderContext prev; // Previous context

    public ReaderContext() {
        fileName = "";
        inputStream = null;
        fileContextId = 0;
        inputLine = "";
        lineNumber = 0;
        isPipe = 0;
        prev = null;
    }
}

File: typescript/base/src/main/vsdk/toolkit/io/context/ReaderContext.ts
import { InputStream } from "../../../../java/io/InputStream";

export class ReaderContext {
  public static readonly MGF_MAXIMUM_INPUT_LINE_LENGTH = 4096;
  public static readonly MGF_MAXIMUM_ARGUMENT_COUNT = (ReaderContext.MGF_MAXIMUM_INPUT_LINE_LENGTH / 4);

  public fileName: string;
  public inputStream: InputStream | null;
  public fileContextId: number;
  public inputLine: string;
  public lineNumber: number;
  public isPipe: number;
  public prev: ReaderContext | null;

  public constructor() {
    this.fileName = "";
    this.inputStream = null;
    this.fileContextId = 0;
    this.inputLine = "";
    this.lineNumber = 0;
    this.isPipe = 0;
    this.prev = null;
  }
}
